File: student-management/src/main/java/model/Course.java
package model;

/**
 * 课程成绩实体类
 * 包含课程名称与对应的考试成绩
 */
public class Course {

    private String name;
    private int score;

    public Course() {
    }

    public Course(String name, int score) {
        this.name = name;
        this.score = score;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public int getScore() {
        return score;
    }

    public void setScore(int score) {
        this.score = score;
    }

    @Override
    public String toString() {
        return name + ": " + score;
    }
}

File: student-management-c/model/course.c
#include "course.h"
#include <string.h>

void course_init(Course *c, const char *name, int score) {
    strncpy(c->name, name, sizeof(c->name) - 1);
    c->name[sizeof(c->name) - 1] = '\0';
    c->score = score;
}
